src/GraphQl/Type/FieldsBuilder.php

```diff
@@ -232,6 +232,8 @@ private function getResourceFieldConfiguration(string $resourceClass, ResourceMe
 
             $args = [];
             if (!$input && null === $mutationName && !$isStandardGraphqlType && $this->typeBuilder->isCollection($type)) {
+                $propertyResourceMetadata = $this->resourceMetadataFactory->create($className);
```

Comment:
Why property? Maybe we could use `$resourceMetadata` and rename the argument in `$parentResourceMetadata`?

Reply:
Because the method parameter for `getResourceFieldConfiguration` is `property`, and that's what this resourceMetadata belongs to - but your naming would also be fine for me 👍

Reply:
But this method is not used necessarily with a property :slightly_smiling_face: It's also used at the top-level.

Reply:
You are right, maybe `$collectionResourceMetadata` would be better?

Reply:
In fact we should even remove the `$resourceMetadata` argument in the method.
And I would move this initialization above the condition like this:
```php
$resourceMetadata = null;
if (!empty($className)) {
    $resourceMetadata = $this->resourceMetadataFactory->create($className);
}
```

Reply:
Looks good to me 👍 - also catches the other usage I've overlooked (in `mergeFilterArgs`) 🙈